### PR TITLE
Add Sentinel governance proposal

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
Here is a demo implementation: https://github.com/ewasm/sentinel-governance

File: sentinel.md [hunk 1]
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
...
+        I32Div,
+        I32Mod,
+        // TBA
+    }
+
+    // Returns the cost for the given kind.
+    function getCost(CostKind kind) external view returns (uint64);
+
+    // Returns all the costs as an array, in order the enum is defined above.
+    function getAllCosts() etxernal view returns (uint64[]);
```
Comment (lrettig) [inline sentinel.md]: typo here in `external`

File: sentinel.md [hunk 2]
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
...
+}
+
+interface Sentinel {
+    // Takes a contract as an input, validates and injects metering statements and returns the upadted version.
```
Comment (lrettig) [inline sentinel.md]: typo here in "updated"

File: sentinel.md [hunk 3]
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
...
+    //
+    // Note, the Sentinel contract should also accept an incoming call with the `0x0061736D` selector and process
```
Comment (lrettig) [inline sentinel.md]: I don't understand this comment. Where does the `0x0061736D` selector come from?

File: sentinel.md [hunk 4]
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
 The process of metering is [explained in its own chapter](./metering.md).
 
 The Sentinel is a system contract which means its code is fixed and is part of the genesis or hardfork ewasm is enabled on.
-As a result, changing or updating it requires a hard fork.
+As a result, changing or updating it requires a hard fork. In order to facilitate a more dynamic ecosystem a proposal follows
+here to introduce upgradability for the Sentinel contract.
+
+## Governance (upgradability)
+
+The Sentinel contract is split into two parts:
```
Comment (lrettig) [inline sentinel.md]: Might be worth mentioning the `SentinelGovernance` contract/interface here as well. It's a little confusing to say there are two parts, then introduce a third part further down.

File: sentinel.md [hunk 5]
```diff
@@ -7,4 +7,85 @@ This system contract performs two important tasks:
...
+
+    // Can be called by anyone. Applies the change to the Sentinel if and only all
+    // governance systems proposed the same address.
+    //
+    // Returns a failure (revert) if the setting failed;
+    function applyChange() external;
+}
+```
+
+A sample implementation of the above exists under the [sentinel-governance](https://github.com/ewasm/sentinel-governance) project.
```
Comment (lrettig) [inline sentinel.md]: `SentinelGovernance` hasn't been implemented there yet, right?

**Comment by lrettig** [comment]
This is awesome! Thanks @axic. Just reviewed and added some comments here.

File: sentinel.md (hunk 3 above)
Comment (axic) [reply sentinel.md]: Wasm header?

Comment (lrettig) [reply sentinel.md]: Aha :) Thanks

Comment (axic) [reply sentinel.md]: It is just a trick to avoid changing Hera at this moment 😉

**Comment by axic** [comment]
@lrettig I think I've addressed all your comments. Please review again (including the entire file because it has changed in a different PR).

Would like to merge this to have a platform changing it with PRs.

**Pull request merged**